app/controllers/summary_controller.rb

```diff
@@ -2,27 +2,40 @@ class SummaryController < ApplicationController
   before_action :authorize
 
   def index
-    client_namespace = current_user.client_slug.titleize
-    @client_summarizer = ClientSummarizer.new(
-      client_namespace: client_namespace,
-      fiscal_year: params[:fiscal_year]
-    )
-    @summary = @client_summarizer.run
+    @summaries = titleized_client_namespaces.map do |cn|
+      get_client_summary(cn, params[:fiscal_year])
+    end
   end
 
   private
 
   def authorize
-    if !adminish_role? || !client_slug?
+    if !current_user.any_admin? || (needs_client_slug? && !client_slug?)
```

Comment:
nice! much better. one question:

for users with `client_admin` role, we are checking that they do not have a client slug. But then in ` titleized_client_namespaces` we return the `client_slug` for the user. What do we want client admins to see and is this accomplishing that?